src/STAT_lmonFrontEnd.C

```diff
@@ -0,0 +1,742 @@
+            if (proctab_[proctabSize_ - 1].pd.host_name == NULL)
+            {
+                printMsg(STAT_ALLOCATE_ERROR, __FILE__, __LINE__, "%s Failed to strdup(%s) to host_name\n", strerror(errno), hostname_);
+                free(remoteNode);
+                return STAT_ALLOCATE_ERROR;
+            }
+        }
+        else
+            proctab_[proctabSize_ - 1].pd.host_name = strdup(remoteNode);
+        free(remoteNode);
+    }
+    else
+    {
+        proctab_[proctabSize_ - 1].pd.host_name = strdup(hostname_);
+        if (proctab_[proctabSize_ - 1].pd.host_name == NULL)
+        {
+            printMsg(STAT_ALLOCATE_ERROR, __FILE__, __LINE__, "%s Failed to strdup(%s) to host_name\n", strerror(errno), hostname_);
+            return STAT_ALLOCATE_ERROR;
+        }
+    }
+    proctab_[proctabSize_ - 1].pd.pid = pid;
+    return STAT_OK;
+}
+
+bool STAT_lmonFrontEnd::checkNodeAccess(const char *node)
+{
+    /* MRNet CPs launched through alps */
+    if (lmonRmInfo_.rm_supported_types != NULL)
+    {
+        if (lmonRmInfo_.rm_supported_types[lmonRmInfo_.index_to_cur_instance] == RC_alps)
+            return true;
+    }
+    char command[BUFSIZE], testOutput[BUFSIZE], runScript[BUFSIZE], checkHost[BUFSIZE], *rsh = NULL, *envValue;
+    FILE *output, *temp;
+
+    /* TODO: this is generally not good... a quick check for clusters where each node has the same hostname prefix */
+    /* A first level filter, compare the first 3 letters */
+    gethostname(checkHost, BUFSIZE);
+    if (strncmp(node, checkHost, 3) != 0 && strcmp(node, "localhost") != 0)
+        return false;
+
+    envValue = getenv("XPLAT_RSH");
+    if (envValue != NULL)
+        rsh = strdup(envValue);
+    if (rsh == NULL)
+    {
+        rsh = strdup("rsh");
+        if (rsh == NULL)
+        {
+             printMsg(STAT_ALLOCATE_ERROR, __FILE__, __LINE__, "%s: failed to set rsh command\n", strerror(errno));
+            return false;
+        }
+    }
+
+    /* Use run_one_sec script if available to tolerate rsh slowness/hangs */
+    snprintf(runScript, BUFSIZE, "%s/bin/run_one_sec", getInstallPrefix());
+    temp = fopen(runScript, "r");
+    if (temp == NULL)
+        snprintf(command, BUFSIZE, "%s %s hostname 2>1", rsh, node);
+    else
+    {
+        snprintf(command, BUFSIZE, "%s %s %s hostname 2>1", runScript, rsh, node);
+        fclose(temp);
+    }
+    output = popen(command, "r");
+    free(rsh);
+    if (output == NULL)
+        return false;
+    fgets(testOutput, sizeof(testOutput), output);
+    pclose(output);
+    if (strcmp(testOutput, "") == 0)
+        return false;
+    return true;
+}
+
+
+int STAT_lmonFrontEnd::getNumProcs()
+{
+    return proctabSize_;
+}
+
+const char* STAT_lmonFrontEnd::getHostnameForProc(int procIdx)
+{
+    return proctab_[procIdx].pd.host_name;
+}
+int STAT_lmonFrontEnd::getMpiRankForProc(int procIdx)
+{
+    return proctab_[procIdx].mpirank;
+}
+
+#ifndef USE_CTI
```

Comment:
Setting the factory at build time.   The ifndef part really should be part of the PR that adds CTI support, but I didn't want to have to revisit this file.     USE_CTI is added to the config in the that subsequent PR.